src/plugins/lineendings/lineendings.c

```diff
@@ -0,0 +1,177 @@
+			return counter;
+	}
+	return NA;
+}
+static int checkLineEndings(const char *fileName, Lineending validLineEnding)
+{
+	FILE *fp;
+	Lineending lineEnding = NA;
+	Lineending found = NA;
+	uint8_t fc, sc;
+	fc = sc = 0;
+	fp = fopen(fileName, "rb");
+	if(fp == NULL)
+	{
+		return -1;
+	}
+	fread(&fc, 1, 1, fp);
+	while(!feof(fp))
+	{
+		fread(&sc, 1, 1, fp);
+		switch(fc)
+		{
+			case LF_BYTE:
+				if(sc == CR_BYTE)
+					found = LFCR;
+				else if(sc == LF_BYTE)
+					found = LF;
+				else if(sc)
+					found = LF;
+				break;
+			case CR_BYTE:
+				if(sc == LF_BYTE)
+					found = CRLF;
+				else if(sc == CR_BYTE)
+					found = CR;
+				else if(sc)
+					found = CR;
+				break;
+		}
+		if(found == CRLF || found == LFCR)
+		{
+			fread(&sc, 1, 1, fp);
+		}
+		if(lineEnding == NA && found != NA)
+		{
+			lineEnding = found;
+			if(validLineEnding != NA && lineEnding != validLineEnding)
+			{
+				fclose(fp);
+				return -2;
+			}  
+			found = NA;
+		}
+		else if(lineEnding != found && found != NA)
+		{
+			fclose(fp);
+			return -3;
+		}
+		fc = sc;
+		found = NA;
+	}
+	fclose(fp); 
+	return 0;
+}
+
+int elektraLineendingsGet(Plugin *handle ELEKTRA_UNUSED, KeySet *returned ELEKTRA_UNUSED, Key *parentKey ELEKTRA_UNUSED)
+{
+	if (!strcmp(keyName(parentKey), "system/elektra/modules/lineendings"))
+	{
+		KeySet *contract = ksNew (30,
+				keyNew ("system/elektra/modules/lineendings",
+					KEY_VALUE, "lineendings plugin waits for your orders", KEY_END),
+				keyNew ("system/elektra/modules/lineendings/exports", KEY_END),
+				keyNew ("system/elektra/modules/lineendings/exports/get",
+					KEY_FUNC, elektraLineendingsGet, KEY_END),
+				keyNew ("system/elektra/modules/lineendings/exports/set",
+					KEY_FUNC, elektraLineendingsSet, KEY_END),			
+#include ELEKTRA_README(lineendings)
+				keyNew ("system/elektra/modules/lineendings/infos/version",
+					KEY_VALUE, PLUGINVERSION, KEY_END),
+				KS_END);
+		ksAppend (returned, contract);
+		ksDel (contract);
+
+		return 1; /* success */
+	}
+	/* get all keys */
+	KeySet *config = elektraPluginGetConfig(handle);
+	Key *valid = ksLookupByName(config, "/valid", 0);
+	Lineending validLineEnding = strToLE(keyString(valid));
+	int ret;
+	ret = checkLineEndings(keyString(parentKey), validLineEnding);
+	if(ret == (-3))
+	{
+			ELEKTRA_SET_ERROR(115, parentKey, "Inconsistent line endings");
+			return -1;
+	}
+	else
+		return 1;
+}
+
+int elektraLineendingsSet(Plugin *handle, KeySet *returned ELEKTRA_UNUSED, Key *parentKey)
+{
+	KeySet *config = elektraPluginGetConfig(handle);
+	Key *valid = ksLookupByName(config, "/valid", 0);
+	Lineending validLineEnding = strToLE(keyString(valid));
+	int ret;
+	ret = checkLineEndings(keyString(parentKey), validLineEnding);
+	switch(ret)
+	{
+		case (-1):
+			ELEKTRA_SET_ERRORF(113, parentKey, "Couldn't open file %s\n", keyString(parentKey));
+			return 1;
+			break;
+		case (-2):
+			ELEKTRA_SET_ERROR(114, parentKey, "Invalid line ending");
```

Comment:
please give more  information (which line ending, and which character or even "line")